Framework/Algorithms/src/DetectorDiagnostic.cpp

```diff
@@ -559,9 +559,11 @@ DetectorDiagnostic::makeMap(API::MatrixWorkspace_sptr countsWS) {
                              "detector to spectra map. Try with LevelUp=0.");
   }
 
-  for (size_t i = 0; i < countsWS->getNumberHistograms(); i++) {
-    detid_t d = (*(countsWS->getSpectrum(i).getDetectorIDs().begin()));
-    auto anc = instrument->getDetector(d)->getAncestors();
+  const SpectrumInfo &spectrumInfo = countsWS->spectrumInfo();
+
+  for (size_t i = 0; i < countsWS->getNumberHistograms(); ++i) {
+
+    auto anc = spectrumInfo.detector(i).getAncestors();
```

Comment:
What is `getAncestors` doing for `DetectorGroup`? Unless it gives the ancestors of the first detector, this changes behavior.